### ISSUE TITLE
socketio not connecting

### ISSUE THREAD

**Issue by pwinston** [pwinston/webmon#4]
Running `webmon.py` by itself, not from napari. When a browser pulls down `/viewer` it seems like some socketio stuff is happening but no socketio handlers are called in JS or in Python.

Webmon should log "connect" and "Creating background thread" and viewer should log "input_data_response" in the browser.

    (napari39) ➜  webmon git:(main) python webmon.py
    INFO:webmon:Logging to console.
    INFO:webmon:Webmon: Starting process 94975
    INFO:webmon:Webmon: args ['webmon.py']
    INFO:webmon:Webmon: Serving http://localhost:5000/ 
    NapariClient: NAPARI_MON_CLIENT not defined
    ERROR:webmon:NapariClient not created.
    (94975) wsgi starting up on http://0.0.0.0:5000
    (94975) accepted ('127.0.0.1', 54640)
    (94975) accepted ('127.0.0.1', 54642)
    127.0.0.1 - - [29/Nov/2020 15:28:15] "GET /viewer HTTP/1.1" 200 432 0.003922
    127.0.0.1 - - [29/Nov/2020 15:28:15] "GET /static/bundle.js HTTP/1.1" 200 671708 0.007435
    127.0.0.1 - - [29/Nov/2020 15:28:15] "GET /socket.io/?EIO=4&transport=polling&t=NOLiHHe HTTP/1.1" 200 371 0.000614
    127.0.0.1 - - [29/Nov/2020 15:28:15] "GET /favicon.ico HTTP/1.1" 404 380 0.000565
    127.0.0.1 - - [29/Nov/2020 15:28:16] "GET /socket.io/?EIO=4&transport=polling&t=NOLiHV3 HTTP/1.1" 200 371 0.000331
    127.0.0.1 - - [29/Nov/2020 15:28:17] "GET /socket.io/?EIO=4&transport=polling&t=NOLiHqz HTTP/1.1" 200 371 0.000289
    127.0.0.1 - - [29/Nov/2020 15:28:22] "GET /socket.io/?EIO=4&transport=polling&t=NOLiJ3A HTTP/1.1" 200 371 0.000391

**Comment by pwinston** [comment]
I hammered the original FlaskTest code into our bundle format and it seems to have the same problem. The log seems to indicate some socketio stuff is happening. But the actual socketio handlers in Python and JS are not running.

**Comment by pwinston** [comment]
I set my socketio version to various older ones in package.json such as:

        "socket.io-client": "~1.4.8",

But then the viewer crashes with:

    bundle.js:formatted:1635 Uncaught ReferenceError: global is not defined
    at bundle.js:formatted:1635

If I turn off minify the error is:

    // node_modules/blob/index.js
    var require_blob = __commonJS((exports,module)=>{
        var BlobBuilder = global.BlobBuilder || global.WebKitBlobBuilder || global.MSBlobBuilder || global.MozBlobBuilder;
        var blob

**Comment by manzt** [comment]
Ah, sorry about that Phil! I checkout `main` and tried what you have above and have produced the same error for `1.4.8`. I tried out two options that _seem_ to be working:

1.) If you want to use `1.4.8`, you can use use the `--define:global=window` with esbuild to replace the `global` with `window` (which is what global-like object in the browser). I _think_ this is a safe replace currently, but I'd worry if other modules use a variable named `global`, things could quietly break and you'd get unexpected behavior (it's really a hack).

```bash
$ esbuild --bundle --define:global=window --outfile=../static/bundle.js src/index.js
```

2.) Use `socket.io-client` 2.x.x. I think from 1 -> 2 the npm `socket.io-client` library module type has better compatibility with bundlers. The example using Webpack [here](https://github.com/socketio/socket.io/tree/2.0.3/examples/webpack-build) uses [`v0.2.0`](https://github.com/socketio/socket.io/blob/2.0.3/examples/webpack-build/package.json). 

```bash
$ cd js && npm install socket.io-client@2
```

**Comment by pwinston** [comment]
No problem. 

I downgraded it to the last 2.X release which was 2.3.1 and that seems to work:

https://github.com/pwinston/webmon/commit/c89c01a7c845ced7558855ef75f21e98dc91471d

So that's a perfectly good solution. Until Flask-SocketIO supported 3.x.

I also set it back to using eventlet but without `eventlet.monkey_patch()`. I have no idea what Flask-SocketIO transport is best. But seems good to standardize on one. So we don't have to deal with bugs that vary based on which one you are using.
